Cohere/CohereClient.cs

```diff
@@ -0,0 +1,180 @@
+        _httpClient = new HttpClient();
+        _httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", _apiKey);
+    }
+
+    /// <summary>
+    /// Initalizes a new instance of the CohereClient class with a custom HttpClient (e.g a mock)
+    /// </summary>
+    /// <param name="apiKey"> The API key to use for requests </param>
+    /// <param name="httpClient"> The custom HttpClient to use for requests </param>
+    public CohereClient(string apiKey, HttpClient httpClient)
+    {
+        _apiKey = apiKey;
+        _httpClient = httpClient;
+        _httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", _apiKey);
+    }
+
+    /// <summary>
+    /// Sends a request to the Cohere API to generate text based on messages provided
+    /// </summary>
+    /// <param name="chatRequest"> The request body sent to Cohere to generate text </param>
+    /// <returns> The response from Cohere as a ChatResponse object </returns>
+    public async Task<ChatResponse> ChatAsync(ChatRequest chatRequest)
+    {
+        ArgumentNullException.ThrowIfNull(chatRequest);
+
+        var response = await SendRequestAsync("chat", chatRequest);
+
+        return (ChatResponse) response;
+    }
+
+    /// <summary>
+    /// Sends a request to the Cohere API to classify text
+    /// </summary>
+    /// <param name="classifyRequest"> The request body sent to Cohere to classify </param>
+    /// <returns> The response from Cohere as a ClassifyResponse object </returns>
+    public async Task<ClassifyResponse> ClassifyAsync(ClassifyRequest classifyRequest)
+    {
+        ArgumentNullException.ThrowIfNull(classifyRequest);
+
+        var response = await SendRequestAsync("classify", classifyRequest);
+
+        return (ClassifyResponse) response;
+    }
+
+    /// <summary>
+    /// Sends a request to the Cohere API to rerank text
+    /// </summary>
+    /// <param name="rerankRequest"> The request body sent to Cohere to rerank </param>
+    /// <returns> The response from Cohere as a RerankResponse object </returns>
+    public async Task<RerankResponse> RerankAsync(RerankRequest rerankRequest)
+    {
+        ArgumentNullException.ThrowIfNull(rerankRequest);
+
+        var response = await SendRequestAsync("rerank", rerankRequest);
+
+        return (RerankResponse) response;
+    }
+
+    /// <summary>
+    /// Private method to send a request to the Cohere API
+    /// </summary>
+    /// <param name="endpoint"> The endpoint to send the request to </param>
+    /// <param name="requestBody"> The request body to send to the API </param>
+    /// <returns> The response from the API as an ICohereResponse object </returns> 
+    private async Task<ICohereResponse> SendRequestAsync(string endpoint, ICohereRequest requestBody)
+    {
+        var serializedRequest = endpoint switch
+        {
+            "chat" => JsonSerializer.Serialize((ChatRequest)requestBody, _jsonSerializerOptions),
```

Comment:
@AnjaliMohammed  - It's a good practice to avoid "magic strings" for several reasons, especially maintainability. If we need to update this value in the future, having it as a magic string means updating every instance manually, with no compiler errors to help the refactor. Could you convert this to an Enum for better reliability?